src/ucs/datastruct/mpool.c

```diff
@@ -27,8 +27,12 @@ static inline ucs_mpool_elem_t *ucs_mpool_chunk_elem(ucs_mpool_data_t *data,
                                                      ucs_mpool_chunk_t *chunk,
                                                      unsigned elem_index)
 {
-    return UCS_PTR_BYTE_OFFSET(chunk->elems,
-                               elem_index * ucs_mpool_elem_total_size(data));
+    ucs_mpool_elem_t *elem =
+        UCS_PTR_BYTE_OFFSET(chunk->elems,
+                            elem_index *
+                            ucs_mpool_elem_total_size(data));
+    VALGRIND_MAKE_MEM_DEFINED(elem, sizeof *elem);
```

Comment:
can we do it only in ucs_mpool_chunk_leak_check?

Reply:
fixed